### PR TITLE
Set visualization width to container width

### LINKED ISSUE #24: Make charts responsive.
**TODO**
- [ ] Store vega view in vega-viewer component internal state for updates.
- [ ] Add resize event listener.
- [ ] Associate the new container dimensions with the vega spec.

### PR DESCRIPTION
**Pull request by okal**
Closes #24

### CODE REVIEW

File: src/vega_viewer/components/vega_viewer.cljs [hunk 1]
```diff
@@ -3,21 +3,52 @@
...
             [sablono.core :refer-macros [html]]))
 
+(defn set-resize-handler
+  [owner]
+  (let [resize-handler
+        (.addEventListener js/window "resize"
+                           (fn [event]
+                             (om/set-state! owner
+                                            :container-width
+                                            (-  (.-clientWidth (om/get-node owner "vega-container"))
```
Comment (pld) [inline src/vega_viewer/components/vega_viewer.cljs]: extra space in `(-  (`

File: src/vega_viewer/components/vega_viewer.cljs [hunk 2]
```diff
@@ -3,21 +3,52 @@
...
+                            (fn [chart]
+                              (let [view (chart #js {:el vega-container})]
+                                (.update view)))))))
+
     om/IDidMount
     (did-mount [_]
+      (set-resize-handler owner)
       (let [vega-container (om/get-node owner "vega-container")
-            vega-spec-as-js (clj->js vega-spec)]
+            container-width (.-clientWidth vega-container)
```
Comment (pld) [inline src/vega_viewer/components/vega_viewer.cljs]: lines 45-51 look like they're the exact same as 28-34

File: src/vega_viewer/components/vega_viewer.cljs (hunk 1 above)
Comment (okal) [inline src/vega_viewer/components/vega_viewer.cljs]: :+1:

File: src/vega_viewer/components/vega_viewer.cljs (hunk 2 above)
Comment (okal) [inline src/vega_viewer/components/vega_viewer.cljs]: They were initially slightly different. I'll pull out the code into a function.

**Comment by okal** [comment]
@pld How's it looking now?

**Comment by pld** [comment]
lgtm!

**Pull request merged**